web/src/engine/websites/ComicZerosum.ts

```diff
@@ -0,0 +1,205 @@
+import { Tags } from '../Tags';
+import icon from './ComicZerosum.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchProto, FetchRequest } from './../FetchProvider';
+
+type APISingleTitle = {
+    title: APITitle
+}
+
+type APITitles = {
+    titles: APITitle[]
+}
+
+type APITitle = {
+    id: number,
+    tag: string,
+    name: string
+}
+
+type TitleView = {
+    title: APITitle,
+    chapters: APIChapter[]
+}
+
+type APIChapter = {
+    id: number,
+    name: string,
+    imageUrl: string
+}
+
+type APIPages = {
+    pages: {
+        imageUrl: string
+    }[]
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private api_url = 'https://api.zerosumonline.com/api/v1/';
+    private protoTypes = `
```

Comment:
Probably better to extract this as constant string outside of the class, maybe even outside of the file and import the constant string from there.

Reply:
Outside of the class : fine.
Outside of the file? you mean not a proto file but a ts file? I wanna make sure its not impacted by _index.ts rebuilding script? 

Couldn't we juste make sure the building process copy the files, and keep proto files as they are? Which help because they are recognized by code editors.

Reply:
I assume you can use any file extension you want for a typescript file, but `.ts` is the only one that is inferred.

_FooBar.proto_
```typescript
export const protoTypeDefinition: string = `
message A ...
message B ...
`;
```

_FooBar.ts_
```typescript
import { protoTypeDefinition } from './FooBar.proto';
const root = proto.parse(protoTypeDefinition);
```

Reply:
If you want a really fancy solution you can write a [tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) method which can directly be used for string literal (templates)

```typescript
export function proto(strings: string[], ...) {
    return proto.parse(strings.join(''))...;
}

const root = proto`
message A ...
message B ...
`;
```